src/bootman/bootman.c

```diff
@@ -446,6 +457,208 @@ bool boot_manager_set_uname(BootManager *self, const char *uname)
         return self->have_sys_kernel;
 }
 
+static bool boot_manager_is_inird_valid_name(const char *name)
+{
+        if (strlen(name) < 3) {
+                return false;
+        }
+
+        return isdigit(name[0]) && isdigit(name[1]);
+}
+
+static int boot_manager_sort_initrd_name(const void *name1, const void *name2)
+{
+        return strcmp((char *)name1, (char *)name2);
+}
+
+bool boot_manager_get_initrd_freestanding(BootManager *self)
+{
+        autofree(DIR) *initrd_dir = NULL;
+        NcArray *dir_files = NULL;
+        struct dirent *ent = NULL;
+        struct stat st = { 0 };
+
+        if (!self || !self->initrd_freestanding_dir) {
+                return false;
+        }
+
+        dir_files = nc_array_new();
+        OOM_CHECK_RET(dir_files, false);
+
+        initrd_dir = opendir(self->initrd_freestanding_dir);
+        if (!initrd_dir) {
+                LOG_ERROR("Error opening %s: %s", self->initrd_freestanding_dir, strerror(errno));
+                nc_array_free(&dir_files, NULL);
+                return false;
+        }
+
+        while ((ent = readdir(initrd_dir)) != NULL) {
+                char *initrd_name = NULL;
+                autofree(char) *path = NULL;
+
+                path = string_printf("%s/%s", self->initrd_freestanding_dir, ent->d_name);
+
+                if (!boot_manager_is_inird_valid_name(ent->d_name)) {
+                        continue;
+                }
+
+                /* Some kind of broken link */
+                if (lstat(path, &st) != 0) {
+                        continue;
+                }
+
+                /* Regular only */
+                if (!S_ISREG(st.st_mode)) {
+                        continue;
+                }
+
+                /* empty files are skipped too */
+                if (st.st_size == 0) {
+                        continue;
+                }
+
+                initrd_name = strdup(ent->d_name);
+                if (!initrd_name) {
+                        DECLARE_OOM();
+                        abort();
+                }
+
+                if (!nc_array_add(dir_files, initrd_name)) {
+                        DECLARE_OOM();
+                        abort();
+                }
+        }
+
+        if (dir_files->len) {
+                nc_array_qsort(dir_files, boot_manager_sort_initrd_name);
+                self->initrd_freestanding = strdup(dir_files->data[0]);
```

Comment:
Wait so there is only one initrd used and depending on the system a list of priortized initrds may be available and we get the smallest numbered one. I'm having a hard time understanding this use case. Shouldn't we be chaining initrds?

Reply:
at this moment systemd-boot don't support multi initrd, but this could change, for that reason we decide to keep this name syntax 00-name to avoid make the users change its initrd name in a future

Reply:
Ugh yes you told me that before. Sorry it has been so long. Right yes that is the most reasonable solution for now and the distro will have to manage this itself.